packages/flutter/test/cupertino/tab_scaffold_test.dart

```diff
@@ -119,6 +120,47 @@ void main() {
     expect(find.text('Page 1'), findsOneWidget);
     expect(find.text('Page 2', skipOffstage: false), isOffstage);
   });
+
+  testWidgets('Last tab gets focus', (WidgetTester tester) async {
+    // 2 nodes for 2 tabs
+    final List<FocusNode> focusNodes = <FocusNode>[new FocusNode(), new FocusNode()];
+
+    await tester.pumpWidget(
+      new WidgetsApp(
+        color: const Color(0xFFFFFFFF),
+        onGenerateRoute: (RouteSettings settings) {
+          return new CupertinoPageRoute<Null>(
+            settings: settings,
+            builder: (BuildContext context) {
+              return new CupertinoTabScaffold(
+                  tabBar: _buildTabBar(),
+                  tabBuilder: (BuildContext context, int index) {
+                    return new TextField(
+                      focusNode: focusNodes[index],
+                      autofocus: true,
+                    );
+                  }
+              );
+            },
+          );
+        },
+      ),
+    );
+
+    expect(focusNodes[0].hasFocus, isTrue);
+
+    await tester.tap(find.text('Tab 2'));
+    await tester.pump();
+
+    expect(focusNodes[0].hasFocus, isFalse);
+    expect(focusNodes[1].hasFocus, isTrue);
+
+    await tester.tap(find.text('Tab 1'));
+    await tester.pump();
+
+    expect(focusNodes[0].hasFocus, isTrue);
+    expect(focusNodes[1].hasFocus, isFalse);
+  });
 }
```

Comment:
another thing worth testing would be what happens if there's a text field outside the tab bar view, and it has the focus, then you switch tabs, does the tab bar view grab the focus or does it stay with the text field outside the tab bar view? It should not change.

Reply:
I may not be understanding the focus mechanism correctly but I tried it in a test in https://github.com/xster/flutter-test/blob/master/t75_navigator_focus/lib/main.dart with the focusing mechanism as is. 

It seems like we don't have a way to yield and give focus diagonally up and across the tree. Once a scope's child node is set to have first focus, I don't see an easy way to signal to relinquish its own or its parents' first focus position to a sibling or parents' siblings. 

I may not be understanding correctly. We can chat in person.

Reply:
I meant a case where the UI is like this:

```
+-----------+
| Field0    |
|+---------+|
||_T1_| T2 ||
|+---------+|
|| Field1  ||
||         ||
|+---------+|
+-----------+
```

Field0 is a focusable field outside the tabs; there's two tabs, in each one there's a focusable field.

Tap on Field0, it takes focus, keyboard comes up; switch to Tab 2 (which shows Field2), does the focus remain in Field0?

Reply:
Chatted with Adam. The existing API doesn't support requesting focus across one layer of ancestry. Left #14792 to augment the API.